patch.crates-io.sh

```diff
@@ -31,17 +31,21 @@ if [[ -f "$toolchain_file" ]]; then
   cp "$toolchain_file" .
 fi
 
-echo "Excluding $solana_dir from workspace"
-echo
-for crate in "${workspace_crates[@]}"; do
-  if grep -q "exclude.*$solana_dir" "$crate"; then
-    echo "$crate is already patched"
-  else
-    cat >> $crate << EXCLUDE
+# only add exclude rule when solana root is under spl root
+spl_project_root=$(git rev-parse --show-toplevel)
```

Comment:
We're using `$(dirname "$0")` higher up, why not just store that directly and grep against that?
```suggestion
project_root=$(dirname "$0")
```
This way, this script can work with other repos, since it's not meant to be spl specific

Reply:
oh! cuz my idea is to compare two full path to know whether solana root is under spl root. if I use`$(dirname "$0")`, I think it will only give me `.` and I couldn't use the same logic 🤔

Reply:
Ah sorry, I meant `pwd` after we cd into `$(dirname "$0")`

Reply:
ah. sorry. I think I have a misunderstand with the situation. I think we can use the pwd!

Reply:
fixed it in [d02e563](https://github.com/solana-labs/solana-program-library/pull/4128/commits/d02e563431b597929f974f72fecb88a75971c081)